src/targets/mobile/scripts/customizeConfigXML.js

```diff
@@ -64,29 +40,31 @@ const customizeConfigXML = function(context) {
     patch * 100 +
     beta
   ).toString()
+  const transformedContent = xsltproc.transform(transformFilePath, configPath, {
+    output: tmpPath
+  })
+  transformedContent.stdout.on('data', function(data) {
+    console.log('xsltproc stdout: ' + data)
+  })
 
-  transformFile = transformFile
-    .toString()
-    .replace('$VERSION_CODE', versionCode)
-    .replace('$ANDROID_VERSION_CODE', androidVersionCode)
-    .replace('$IOS_VERSION_CODE', overrideConfig.fullVersion)
-    .replace('$USER_AGENT_VERSION', 'io.cozy.banks.mobile-' + versionCode)
+  transformedContent.stderr.on('data', function(data) {
+    console.log('xsltproc stderr: ' + data)
+  })
 
-  libxslt.parse(transformFile, (err, stylesheet) => {
-    if (err) {
-      handleError(err)
-    }
+  transformedContent.on('exit', function(code) {
+    if (code === 0) {
+      const config = fs.readFileSync(tmpPath, 'utf8')
 
-    stylesheet.apply(config, (err, output) => {
-      if (err) {
-        handleError(err)
-      }
+      let output = config
+        .toString()
+        .replace('$VERSION_CODE', versionCode)
+        .replace('$ANDROID_VERSION_CODE', androidVersionCode)
+        .replace('$IOS_VERSION_CODE', overrideConfig.fullVersion)
+        .replace('$USER_AGENT_VERSION', 'io.cozy.banks.mobile-' + versionCode)
 
       fs.writeFileSync(configPath, output)
-      console.log(
-        `config.xml successfully transformed using ${transformFilePath}`
-      )
-    })
+    }
+    fs.unlinkSync(tmpPath)
```

Comment:
The temp file is not unlinked if there is a problem. It is not particularly important here but when there are resources, it is good to have the habit of 

```
initializeResource()
try {
 doStuffThatMightThrow()
} finally {
  destroyResource()
}
```

In python, this is built into the language with the `with` statement. A similar approach could be used in JS.

```
withTempFile(tempFile => {
 // do stuff
})
```

Reply:
Is it better now?

Reply:
Yes :) Thanks !